projectq/ops/_basics.py

```diff
@@ -262,12 +272,19 @@ def __eq__(self, other):
             return True
         return False
 
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
     def __str__(self):
-        raise NotImplementedError('This gate does not implement __str__.')
+        return("MatrixGate(" + str(self.matrix.tolist()) + ")")
+
+    def __hash__(self):
+        return hash(str(self))
```

Comment:
Again, this seems superflous because `__hash__()` is inherited.

Reply:
👍

Reply:
uups. Python3 seems to remove __hash__ if __eq__ is redefined...